python/llm/src/bigdl/llm/transformers/low_bit_linear.py

```diff
@@ -520,6 +520,12 @@ def __init__(self, input_features, output_features, qtype, bias=True,
         self.mp_group = mp_group
         self.compute_dtype = None  # only for training
         self.enable_xetla = enable_xetla
+        # empty_cache (by default on arc) for models with large vocabulary (e.g. baichuan/qwen)
+        # for long input sequences.
+        self.low_memory_mode = \
+            self.weight._shape[0]*self.weight._shape[1] >= 70000*4096 and \
+            (get_xpu_device_type(self.weight.data) == "arc" or
```

Comment:
this will be false when the layer is first created

Reply:
correct. Updated. Will detect the device at the first forward and do only once.